core/src/main/java/org/springframework/security/authorization/AuthorityReactiveAuthorizationManager.java

```diff
@@ -48,8 +48,8 @@
 				.flatMapIterable(Authentication::getAuthorities)
 				.map(GrantedAuthority::getAuthority)
 				.any(this.authorities::contains)
-				.map(AuthorizationDecision::new)
-				.defaultIfEmpty(new AuthorizationDecision(false));
+				.map((granted) -> ((AuthorizationDecision) new AuthorityAuthorizationDecision(granted, this.authorities)))
```

Comment:
Given my other comment about changing this constructor to take a collection of `GrantedAuthority`s, I think that this class should store a collection of `GrantedAuthority`s.